src/main/java/lotto/domain/LottoGroup.java

```diff
@@ -11,6 +17,38 @@ public LottoGroup(List<Lotto> lottoGroup) {
     }
 
     public List<Lotto> getLottoGroup() {
-        return lottoGroup;
+        return Collections.unmodifiableList(lottoGroup);
+    }
+
+    public Map<Rank, Integer> matchRank(WinningLotto winningLotto) {
+        Map<Rank, Integer> ranks = new HashMap<>();
+        for (Lotto lotto : lottoGroup) {
+            Rank rank = winningLotto.findRank(lotto);
+            int rankCount = ranks.getOrDefault(rank, 0);
+            ranks.put(rank, rankCount + 1);
+        }
+        return ranks;
+    }
```

Comment:
```suggestion
    public LottoResult matchRank(WinningLotto winningLotto) {
        Map<Rank, Integer> ranks = new HashMap<>();
        for (Lotto lotto : lottoGroup) {
            Rank rank = winningLotto.findRank(lotto);
            int rankCount = ranks.getOrDefault(rank, 0);
            ranks.put(rank, rankCount + 1);
        }
        return new LottoResult(ranks);
    }
```

객체를 만들어서 리턴해주는 방법도 있습니다! :)